WordPress/src/androidTest/java/org/wordpress/android/e2e/EditorTests.java

```diff
@@ -0,0 +1,45 @@
+package org.wordpress.android.e2e;
+
+import android.support.test.rule.ActivityTestRule;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.wordpress.android.e2e.components.MasterbarComponent;
+import org.wordpress.android.e2e.components.SnackbarComponent;
+import org.wordpress.android.e2e.pages.EditorPage;
+import org.wordpress.android.e2e.pages.MySitesPage;
+import org.wordpress.android.support.BaseTest;
+import org.wordpress.android.ui.WPLaunchActivity;
+
+import static org.wordpress.android.BuildConfig.E2E_DEFAULT_USER_SITE_ADDRESS;
+
+public class EditorTests extends BaseTest {
+    @Rule
+    public ActivityTestRule<WPLaunchActivity> mActivityTestRule = new ActivityTestRule<>(WPLaunchActivity.class);
+
+    @Test
+    public void testPublishPost() {
+        String title = "Title";
+        String content = "Content";
+
+        wpLogin();
+
+        MasterbarComponent mb = new MasterbarComponent().goToMySitesTab();
+        sleep();
+        mb.clickBlogPosts();
+
+        new MySitesPage()
+                .startNewPost(E2E_DEFAULT_USER_SITE_ADDRESS);
+
+        new EditorPage()
+                .enterTitle(title)
+                .enterContent(content)
+                .publishPost();
+
+        // Unfortunately it seems to take quite a while to upload, this shouldn't be necessary once we're mocking
+        sleep(10000);
```

Comment:
As you noted this should be resolved once we're mocking the network requests, but for now this sleep is causing the test to fail for me. The upload happens faster on my end and so by the time the test moves on the snackbar is already gone from the screen.

I'm hesitant to spend too much time on fixing this as-is since we plan to set up mocking before we start really using these tests, though.

Reply:
This is something that could be resolved by some of the helpers in https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/support/WPScreenshotSupport.java

It may be worth our while to bring this class into the wider tests, and use these high-level methods to accomplish what we're looking to do. 

Within `EditorPage`, as a for-instance, you could add a `waitForSnackBarToAppear()` method that calls out to `waitForElementToBeDisplayed()`.

I'd be interested to know if it's worth using those sort of high-level wrappers as our own wrapper around Espresso in situations where it doesn't meet our needs?

Reply:
> It may be worth our while to bring this class into the wider tests, and use these high-level methods to accomplish what we're looking to do.

I agree, My moving it out its previous folder was my first step towards this, I was considering moving the functions inside of it into the `BaseTest` class maybe or a parent class for the Page objects.

Reply:
> Within `EditorPage`, as a for-instance, you could add a `waitForSnackBarToAppear()` method that calls out to `waitForElementToBeDisplayed()`.

🤔you're probably right, let me give this a go

Reply:
@rachelmcr @jkmassel I've made some changes, mind taking another look?